ElectromagneticModels/coupledElectromagneticModels/coupledCurrentDensity/coupledCurrentDensityFvPatchVectorField.C

```diff
@@ -0,0 +1,232 @@
+/*---------------------------------------------------------------------------*\
+  =========                 |
+  \\      /  F ield         | OpenFOAM: The Open Source CFD Toolbox
+   \\    /   O peration     | Website:  https://openfoam.org
+    \\  /    A nd           | Copyright (C) 2011-2023 OpenFOAM Foundation
+     \\/     M anipulation  |
+-------------------------------------------------------------------------------
+License
+    This file is part of OpenFOAM.
+
+    OpenFOAM is free software: you can redistribute it and/or modify it
+    under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    OpenFOAM is distributed in the hope that it will be useful, but WITHOUT
+    ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+    FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+    for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with OpenFOAM.  If not, see <http://www.gnu.org/licenses/>.
+
+\*---------------------------------------------------------------------------*/
+
+#include "coupledCurrentDensityFvPatchVectorField.H"
+#include "coupledElectricPotentialFvPatchScalarField.H"
+#include "electromagneticModel.H"
+#include "volFields.H"
+#include "fvPatchFieldMapper.H"
+#include "mappedPatchBase.H"
+#include "addToRunTimeSelectionTable.H"
+
+// * * * * * * * * * * * * * Protected Member Functions  * * * * * * * * * * //
+
+void Foam::coupledCurrentDensityFvPatchVectorField::getThis
+(
+    tmp<scalarField>& sigma
+) const
+{
+    const electromagneticModel& em =
+        patch().boundaryMesh().mesh()
+       .lookupType<electromagneticModel>();
+
+    sigma = em.sigma(patch().index());
+}
+
+void Foam::coupledCurrentDensityFvPatchVectorField::initCoupledPotential()
+{
+    //Allow for single initialization
+    if (!coupled_)
+    {
+        const electromagneticModel& em =
+            patch().boundaryMesh().mesh()
+        .lookupType<electromagneticModel>();
+
+        ePotName_ = em.getCoupledPotentialName(internalField().name());
+        coupled_ = true;
+    }
+}
+
+void Foam::coupledCurrentDensityFvPatchVectorField::add
+(
+    tmp<scalarField>& result,
+    const tmp<scalarField>& field
+) const
+{
+    if (result.valid())
```

Comment:
a comment would help. Not clear when we add, when we replace and when we clone.

Other thing which is probably just my lack of knowledge in OF, what is  the type tmp? I wonder how it is some field which is also callable.